### DISCUSSION TITLE
【求助】IDM 下载自定义 URL 提示 SSL connect error 5

### DISCUSSION THREAD

**Discussion by youyou-sudo** [AlistGo/alist#5682] [Q&A] [answered]
## 环境
Alist版本：3.29.1，docker部署
onedrive app 方式挂载
使用 [alist-proxy.js](https://github.com/alist-org/alist-proxy/blob/main/alist-proxy.js) 文件部署了 workers 并用路由（自定义主机名也试过）方式添加域名
然后我把 URL 填入存储的下载 URL
cloudflare 设置项： SSL 加密设置为灵活，自动 HTTPS 重写，始终使用 HTTPS（这些开关都测试过，无果）
## 问题
这个下载 URL，在不挂代理的情况下，在手机浏览器、手机 IDM，电脑浏览器都能正常下载
但是电脑端 IDM 提示 SSL connect error 5
![image](https://github.com/alist-org/alist/assets/60727068/92a8e318-30d9-48dd-b36b-d6dd3ecd3242)
如果我挂上代理，电脑的 IDM 就又能正常下

## 一些尝试
- 设置 CF SSL 加密为灵活、完全、完全（严格）
- 使用其他域名通过 [自定义域] 、 [路由] 方式添加到 Worker 
- 检查过 IDM 代理设置，CF 的 SSL 加密设置的是灵活
- 排除电脑问题，用朋友的电脑的 IDM 下载问题依旧
- IDM 设置，下方三项都试过（无果）
![image](https://github.com/alist-org/alist/assets/60727068/68032ed0-6143-48a5-aba2-5cf28744edb8)

Google 过也问过 GPT，现在大脑一片空，不知道什么问题，在此求助各位大佬了
![image](https://github.com/alist-org/alist/assets/60727068/ae7f0445-db92-43c3-b18a-5a2b5b0b14c2)

**Comment by youyou-sudo** [comment, marked as answer]
研究许久，最终 Telegram 群友提示，使用的 `eu.org` 域名被 CFW 检测，处于半墙状态，糟糕的是貌似除了挂代理与换域名才能有效解决此问题
不过这也能解释出现挂代理下载正常，而使用中国网络遇到SSL connect error 5问题

**Comment by jiurou** [comment]
我是在mediafire我网盘下载时才出现这个报错，也不知道该怎么办

**Comment by 775852123** [comment]
C:\Windows\System32\drivers\etc 目录下记事本打开hosts，添加一条 eu.org  78.194.169.74 保存，问题解决。
如果ip地址如有变动，打开  https://www.itdog.cn/batch_tcping/  输入 eu.org 得到测试结果的ip替换即可。

**Reply by youyou-sudo** [reply]
> C:\Windows\System32\drivers\etc 目录下记事本打开hosts，添加一条 eu.org 78.194.169.74 保存，问题解决。 如果ip地址如有变动，打开 https://www.itdog.cn/batch_tcping/ 输入 eu.org 得到测试结果的ip替换即可。

谢谢回复，这也是一个很好的方案
